pages/calendar.js

```diff
@@ -0,0 +1,99 @@
+import { Calendar, momentLocalizer, Views } from "react-big-calendar";
+import moment from "moment";
+import LayoutStyle from "../components/LayoutStyle";
+import back from "../public/images/back.svg";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import styled from "styled-components";
+import { useMemo } from "react";
+import { getSession } from "next-auth/react";
+import Task from "../models/Task";
+import dbConnect from "../lib/dbConnect";
+
+export const getServerSideProps = async (context) => {
+  const session = await getSession(context);
+  if (!session) {
+    return {
+      redirect: {
+        destination: "/onboarding",
+      },
+    };
+  }
```

Comment:
its very good that you use an early return here and only connect to the db after a successful session check 👍